libraries/AP_OpticalFlow/AP_OpticalFlow.cpp

```diff
@@ -19,38 +19,38 @@ extern const AP_HAL::HAL& hal;
 
 #ifndef OPTICAL_FLOW_TYPE_DEFAULT
  #if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_CHIBIOS_SKYVIPER_F412 || defined(HAL_HAVE_PIXARTFLOW_SPI)
-  #define OPTICAL_FLOW_TYPE_DEFAULT OpticalFlowType::PIXART
+  #define OPTICAL_FLOW_TYPE_DEFAULT Type::PIXART
  #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BEBOP
-  #define OPTICAL_FLOW_TYPE_DEFAULT OpticalFlowType::BEBOP
+  #define OPTICAL_FLOW_TYPE_DEFAULT Type::BEBOP
  #else
-  #define OPTICAL_FLOW_TYPE_DEFAULT OpticalFlowType::NONE
+  #define OPTICAL_FLOW_TYPE_DEFAULT Type::NONE
  #endif
 #endif
 
-const AP_Param::GroupInfo OpticalFlow::var_info[] = {
+const AP_Param::GroupInfo AP_OpticalFlow::var_info[] = {
     // @Param: _TYPE
     // @DisplayName: Optical flow sensor type
     // @Description: Optical flow sensor type
     // @Values: 0:None, 1:PX4Flow, 2:Pixart, 3:Bebop, 4:CXOF, 5:MAVLink, 6:DroneCAN, 7:MSP, 8:UPFLOW
     // @User: Standard
     // @RebootRequired: True
-    AP_GROUPINFO_FLAGS("_TYPE", 0,  OpticalFlow,    _type,   (int8_t)OPTICAL_FLOW_TYPE_DEFAULT, AP_PARAM_FLAG_ENABLE),
+    AP_GROUPINFO_FLAGS("_TYPE", 0,  AP_OpticalFlow,    _type,   (float)OPTICAL_FLOW_TYPE_DEFAULT, AP_PARAM_FLAG_ENABLE),
```

Comment:
This change to cast to a float seems odd to me because the parameter is an AP_Int8